docs/api.md

```diff
@@ -1273,7 +1273,7 @@ or updates existing device details for this user/session
 If no device `id` is specified,
 both `name` and `type` must be provided.
 If a device `id` is specified,
-at least one of `name`, `type`, `pushCallback` and `pushPublicKey`
+at least one of `name`, `type`, `pushCallback`, `pushPublicKey` and `pushAuthKey`
```

Comment:
I think we should do stricter validation on the (callback, pubkey, authkey) triple here, to guard against getting into unusable states.  Some options:

* If you specify *any* of these three options, you must specify all three of them
* You can specify either a `pushCallback` without keys, or a full `pushCallback`, `pushPublicKey`, `pushAuthKey` set, but not any other subset of these values.

@eoger thoughts?  I've no idea how hard it would be to enforce those conditions in the validation layer.